### DISCUSSION TITLE
Using the EventSubscriberInterface in Symfony

### DISCUSSION THREAD

**Discussion by Evoolo** [brefphp/bref#1380] [Support] [answered]
Hello, 

I am trying to subscribe to ControllerArgumentsEvent. this works great in a development environment but unfortunately not in Lambda. There the event subscribers are ignored. Can you please tell me why this happens? Do I need to handle event subscribers differently in bref?

Thank you for your help! 

```
class RequestLimitCheckSubscriber implements EventSubscriberInterface
{
    private ManagerRegistry $managerRegistry;

    public function __construct(ManagerRegistry $managerRegistry, LoggerInterface $logger)
    {
        $this->managerRegistry = $managerRegistry;
        $this->logger = $logger;
        
    }

    public static function getSubscribedEvents(): array
    {
        return [
            ControllerArgumentsEvent::class => 'onControllerEvent'
        ];
    }

    private function checkRequestLimit($event)
    {
        $em = $this->managerRegistry->getManager();
        $apiToken = $event->getRequest()->headers->get('Authorization');

        $request = $event->getRequest();
        $account = $em->getRepository(Account::class)->findOneBy(['email' => $decodedJwtToken["email"]]);
        
        $subscription = $account->getSubscription();
        $plan = $account->getSubscription()->getPlan();

        $requestsUsed = $subscription->getRequestsUsed();
        $requestsLimit = $plan->getRequestsLimit();

        if ($requestsUsed >= $requestsLimit) {
            throw new HttpException(422, 'Error! Your requests exceeded. Please renew your plan or contact the support.');
        } else {
            return false;
        }
        
    }

    public function onControllerEvent(ControllerArgumentsEvent $event)
    {
        $this->checkRequestLimit($event);
        return true;
    }
   
}
```

**Comment by wysow** [comment, marked as answer]
Hello, I think this kind of subscriber is run in the kernel terminate process, which can't run on lambda because the virtual machine kills the process immediately after the response is sent.

**Reply by Evoolo** [reply]
Thanks for your answer! Do you have any idea how I can work around this limitation? I want to limit the user requests and therefore I have to check before execution if he is authorized to do so and increase a counter in the database after execution.

**Reply by Evoolo** [reply]
Hello again. I looked in the Symfony documentation and saw that the events I'm tapping don't run inside the termination event. 

I subscribe to the events: 
KernelEvents::REQUEST (1)
KernelEvents::CONTROLLER_ARGUMENTS (4)
KernelEvents::RESPONSE (7)

which all take place before the response is played back. 


![alt text](https://symfony.com/doc/current/_images/components/http_kernel/http-workflow-exception.svg)

**Reply by wysow** [reply]
OK so I suggest you to add some logs everywhere to be able to see what's going on...

**Reply by Evoolo** [reply]
You were right. I added some logging and found an issue. So on bref side everything was fine. Thanks for your help!
